### PR TITLE
Closes #5207 on redirect loop with WP on sub-dir

### LINKED ISSUE #5207: Potential redirect loop when using 3.11.4, trailing slash redirection
**Before submitting an issue please check that you’ve completed the following steps:**
- Made sure you’re on the latest version - 3.11.4
- Used the search feature to ensure that the bug hasn’t been reported before

**Describe the bug**
When permalinks are set to be without `/` and website is in the subdirectory we might experience redirect loop. 

**To Reproduce**
1. Use website placed in subdirectory (could be multisite)
2. Change the permalinks to be without `/`
3. Visit frontend in incognito

**Expected behavior**
No redirect loop

**Additional context**
Still not sure if there are any other factors needed to reproduce.

**Temporary solution**
Change the permalinks settings to be with `/`

**Backlog Grooming (for WP Media dev team use only)**
- [ ] Reproduce the problem
- [ ] Identify the root cause
- [ ] Scope a solution
- [ ] Estimate the effort

### PR DESCRIPTION
**Pull request by jeawhanlee**
## Description

This PR fixes the issue with a redirect loop on sub-dirs

Fixes #5207 

## Type of change
- [x] Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

- Tested on TLD and Sub-dir installation on Apache.
- Tested only on TLD installation and multilanguage on NGINX.

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

**Review by CrochetFeve0251 — APPROVED** [review]

**Comment by piotrbak** [comment]
I can see this solution failing for subdirectory multisite on the sites that are not main. Example:
https://subdir.rocketlabsqa.ovh/test2/

Moreover, we should take care of not redirecting 404 errors of static assets. Example:
https://domain.com/idontexist.txt

It should not be redirected to: https://domain.com/idontexist.txt/

**Review by CrochetFeve0251 — APPROVED** [review]

**Review by remyperona — APPROVED** [review]

**Comment by vmanthos** [comment]
@jeawhanlee 

While testing the PR I found and @Mai-Saad confirmed that there is a **redirect loop** for the homepage on a **Single site** installed in a **subfolder** where the permalinks don't have a trailing `/` and the URL includes:
* a cached `OR`
* an ignored query string `OR` 
* a combination of a cached and ignored query string

File: inc/classes/Buffer/class-cache.php [hunk 1]
```diff
@@ -677,15 +700,33 @@ protected function reset_lowercase( $matches ) {
 	 * @return void
 	 */
 	private function maybe_redirect_with_trailing_slash() {
+
+		if ( defined( 'MULTISITE' ) && true === MULTISITE ) {
```
Comment (CrochetFeve0251) [inline inc/classes/Buffer/class-cache.php:704]: Based on @piotrbak on Slack we need also too check `SUBDOMAIN_INSTALL` is set to false

**Review by CrochetFeve0251 — CHANGES_REQUESTED** [review]

File: inc/classes/Buffer/class-cache.php (hunk 1 above)
Comment (piotrbak) [reply inc/classes/Buffer/class-cache.php:704]: We seem to be fine on subdomains. We should let QA confirm thag

**Review by CrochetFeve0251 — APPROVED** [review]

**Comment by piotrbak** [comment]
@jeawhanlee @CrochetFeve0251 this one can be closed in favour of https://github.com/wp-media/wp-rocket/pull/5215 right?

**Pull request closed without merging**